client/interface.go

```diff
@@ -1,35 +1,17 @@
 package client
 
 import (
-	"context"
-	"time"
-
+	clientinterface "github.com/drand/drand/client/interface"
 	"github.com/drand/drand/log"
 )
 
 // Client represents the drand Client interface.
-type Client interface {
-	// Get returns a the randomness at `round` or an error.
-	// Requesting round = 0 will return randomness for the most
-	// recent known round, bounded at a minimum to the `RoundAt(time.Now())`
-	Get(ctx context.Context, round uint64) (Result, error)
-
-	// Watch returns new randomness as it becomes available.
-	Watch(ctx context.Context) <-chan Result
-
-	// RoundAt will return the most recent round of randomness that will be available
-	// at time for the current client.
-	RoundAt(time time.Time) uint64
-}
+type Client = clientinterface.Client
```

Comment:
What's the reasoning for having this alias instead of moving the interface at this level ? That's usually how go packages can be organized: interface at the top level and implementations in sub folders.

Reply:
The thought is that a normal consumer of drand would depend on `/client` - any client will need the concrete constructor in this package.
The `Interface` here is to prevent circular dependencies, by allow internal packages like the relay to know the interface without caring about the concrete implementation passed.

Reply:
I think I'm saying there shouldn't be any circular dependency, that's something wrong in the design first if there is + alias are not meant for this usage, it's meant for [updating/refactoring codebase](https://talks.golang.org/2016/refactor.article) search "alias"); but if we're changing everything already we shouldn't use it. 
The regular scheme is :
```
toplevelinterface -> client/
implementation -> client/grpc
usage -> relay/ (will import both path above for using and instantiating respectively)
```
Why is that not possible ?  
 
Simplest example I found in the Go lib about this design is for the [hash function](https://golang.org/pkg/crypto/#Hash).

Reply:
Hopefully this works? client is now the interface, and the implementation is in `client/basic`, paralleling the structure for interface + concrete implementation found in libp2p